src/sgame/botlib/bot_nav.cpp

```diff
@@ -426,9 +427,9 @@ void G_BotAddObstacle( const glm::vec3 &mins, const glm::vec3 &maxs, int obstacl
 		// offset mins down by agent height so obstacles placed on ledges are handled correctly
 		tempBox.mins[ 1 ] -= params->walkableHeight;
 
-		nav->cache->addBoxObstacle( tempBox.mins, tempBox.maxs, &ref );
-		obstacleHandles[i] = ref;
+		nav->cache->addBoxObstacle( tempBox.mins, tempBox.maxs, &handles[i] );
 	}
+	obstacleHandles[obstacleNum] = handles;
```

Comment:
Sorry to keep coming up with new stuff, but I think it would be good to use the map `insert` function here and show a warning if there is already a registered obstacle for the given entity since that would lead to a leak.

Or maybe you could just have BotAddObstacle call BotRemoveObstacle so that leaks can't happen.

Reply:
It's not exactly pretty with .insert, but added that